packages/compass-components/src/components/modals/modal-header.tsx

```diff
@@ -77,8 +77,7 @@ function UnthemedModalHeader({
           <Icon glyph="Warning" fill={palette.red.base} role="presentation" />
         </div>
       )}
-
-      <h1 className={cx(titleStyle, darkMode && titleStyleDark)}>{title}</h1>
+      <h1 className={cx(titleStyle, darkMode && titleStyleDark)} data-testid="modal-title" id="modal-title">{title}</h1>
```

Comment:
id here is for area-labelledby in the settings sidebar. Only used there, but I think that's OK. Allows it to just be a normal form modal.